framework/core/js/src/common/extenders/Routes.ts

```diff
@@ -0,0 +1,36 @@
+import Application, { FlarumGenericRoute } from '../Application';
+import ExtenderInterface, { ExtensionModuleInterface } from './ExtenderInterface';
+
+type HelperRoute = (...args: any) => string;
+
+export default class Routes implements ExtenderInterface {
+  private routes: Record<string, FlarumGenericRoute> = {};
+  private helpers: Record<string, HelperRoute> = {};
+
+  /**
+   * Add a mithril route to the application.
+   *
+   * @param name The name of the route.
+   * @param path The path of the route.
+   * @param component must extend `Page` component.
+   */
+  add(name: string, path: `/${string}`, component: any): Routes {
+    this.routes[name] = { path, component };
+
+    return this;
+  }
+
+  helper(name: string, callback: HelperRoute): Routes {
```

Comment:
Is the name "helper" consistent with the concrete implementations included in core's routing helpers? Can we somehow refer to that, even if the type doesn't adds any additional structure or safety, if only for documentation?

Reply:
then how about just `route` instead of `helper`